### PR TITLE
CMR-9255, CMR-9256 - adding non umm-json (ECHO10, DIF10, etc.) support for collection drafts.

### PR DESCRIPTION
**Pull request by eereiter**

### CODE REVIEW

File: indexer-app/src/cmr/indexer/data/concept_parser.clj [hunk 1]
```diff
@@ -57,11 +58,26 @@
...
+(doseq [concept-type concepts/get-draft-concept-types-array]
+  (defmethod parse-concept concept-type
+    [context concept]
+    ;; If the draft record is a json record, then parse it,
+    ;; otherwise figure out what the draft record concept type is
+    ;; and let the parsing code parse it to umm-c.
+    (if (string/includes? (:format concept) "json")
+      (json/parse-string (:metadata concept) true)
+      (let [concept-type (:concept-type concept)]
+        (if (concepts/is-draft-concept? concept-type)
```
Comment (eudoroolivares2016) [inline indexer-app/src/cmr/indexer/data/concept_parser.clj:75]: Should we pull this `(if (concepts/is-draft-concept? concept-type)` to the top of the function to check if the concept is a draft type before this check: `(if (string/includes? (:format concept) "json")`

Comment (eudoroolivares2016) [inline indexer-app/src/cmr/indexer/data/concept_parser.clj:75]: Should we move this `(if (concepts/is-draft-concept? concept-type)` to the top of the function before the `(if (string/includes? (:format concept) "json")` check?

File: system-int-test/test/cmr/system_int_test/search/generics_test.clj [hunk 2]
```diff
@@ -358,3 +360,91 @@
...
+            body (json/parse-string (:body post-results) true)
+            concept-id (:concept-id body)
+            revision-id (:revision-id body)
+            plural-concept-type-name (inf/plural concept-type-string)]
+        (index/wait-until-indexed)
+
+        (testing "Check that test the document ingested before going forward with tests"
+          (is (= 201 (:status post-results)) "failed to ingest test record"))
+
+        (testing "Testing generics name parameter search"
```
Comment (eudoroolivares2016) [inline system-int-test/test/cmr/system_int_test/search/generics_test.clj:387]: Is this more fitting; as I understand drafts are generics but, this is really draft collections
```suggestion
        (testing "Testing draft collections name parameter search"
```

**Review by eudoroolivares2016 — APPROVED** [review]

File: system-int-test/test/cmr/system_int_test/search/generics_test.clj [hunk 3]
```diff
@@ -358,3 +360,91 @@
 
                    "explicit UMM JSON version through suffix"
                    (format "application/vnd.nasa.cmr.umm+json;version=%s" (generics/current-generic-version (keyword concept-type-string))))))))))
+
+;; Test that generic draft collections non json records can be ingested and searched.
+(deftest all-generic-non-json-collection-draft-search-results-test
```
Comment (eudoroolivares2016) [inline system-int-test/test/cmr/system_int_test/search/generics_test.clj:365]: Should we have a `draft_util.clj` namespace for these tests?

**Review by TylerHeald1 — APPROVED** [review]

File: indexer-app/src/cmr/indexer/data/concept_parser.clj (hunk 1 above)
Comment (eereiter) [reply indexer-app/src/cmr/indexer/data/concept_parser.clj:75]: Drafts can have both json or xml records, while right now non drafts are just json, but that can change.  Regardless, if I change the if statement, then I will need to add an extra if to have the same logic.

(if (concepts/is-draft-concept? concept-type)
    (if (string/includes? (:format concept) "json")
      (json/parse-string (:metadata concept) true)
      (let [draft-concept-type (concepts/get-concept-type-of-draft concept-type)
            concept (assoc concept :concept-type draft-concept-type)]
        (umm/parse-metadata context concept)))
    (if (string/includes? (:format concept) "json")
      (json/parse-string (:metadata concept) true)
      (umm/parse-metadata context concept)))

I think it is simpler leaving it as is.

File: system-int-test/test/cmr/system_int_test/search/generics_test.clj (hunk 3 above)
Comment (eereiter) [reply system-int-test/test/cmr/system_int_test/search/generics_test.clj:365]: I don't think we need a draft_util.clj.  We don't have any specific functions other than the test itself.

File: system-int-test/test/cmr/system_int_test/search/generics_test.clj (hunk 2 above)
Comment (eereiter) [reply system-int-test/test/cmr/system_int_test/search/generics_test.clj:387]: Yes, while the record that is being searched is a collection-draft, the search parameter is still a generic search parameter search.  I have changed the comment.

**Pull request merged**